src/v3/webpack.dev.config.ts

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import fs from 'fs';
+import { resolve } from 'path';
+
+import nodemon from 'nodemon';
+import webpack from 'webpack';
+import { merge } from 'webpack-merge';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+
+import baseConfig from './webpack.common.config';
+
+import type { Configuration } from 'webpack';
+import type { Configuration as DevServerConfiguration } from 'webpack-dev-server';
+
+const DEV_SERVER_PORT = 3000;
+const MOCK_SERVER_PORT = 3030;
+
+const WIDGET_RC_JS = resolve(__dirname, '../..', '.widgetrc.js');
+const PLAYGROUND = resolve(__dirname, '../..', 'playground');
+const TARGET = resolve(__dirname, '../..', 'target');
+const ASSETS = resolve(__dirname, '../..', 'assets');
+
+// run `OKTA_SIW_HOST=0.0.0.0 yarn start --watch` to override the host
+const HOST = process.env.OKTA_SIW_HOST || 'localhost';
+const STATIC_DIRS = [PLAYGROUND, TARGET, ASSETS];
+
+const headers = (() => {
+  if (!process.env.DISABLE_CSP) {
+    // Allow google domains for testing recaptcha
+    const scriptSrc = `script-src http://${HOST}:${DEV_SERVER_PORT} https://www.google.com https://www.gstatic.com`;
+    const styleSrc = `style-src http://${HOST}:${DEV_SERVER_PORT} 'nonce-playground'`;
+
+    return {
+      'Content-Security-Policy': `${scriptSrc}; ${styleSrc};`,
+    };
+  }
+  return;
+})();
+
+if (!fs.existsSync(WIDGET_RC_JS)) {
+  // create default WIDGET_RC if it doesn't exist to simplifed the build process
+  fs.copyFileSync(resolve(__dirname, '../..', '.widgetrc.sample.js'), WIDGET_RC_JS);
+}
+
+const devConfig: Configuration = merge(
+  baseConfig,
+  {
+    mode: 'development',
+    devtool: 'source-map',
+    entry: {
+      playground: {
+        import: [
+          `${PLAYGROUND}/main.ts`,
```

Comment:
maybe consider add a minimal playground for IE11, so we don't need to bother with playground IE11 compat issue (not important)

Reply:
The playground actually works in IE11 now since we are using this same webpack config with the playground entrypoint, so the same babel config that makes the widget bundle IE11 compatible are also applied to the playground now. This was a big change that webpack5 made to support multiple entry points much better so we can use this consolidated configuration to build the playground and the widget at the same time.